cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/EquipmentExport.java

```diff
@@ -393,14 +394,22 @@ private static void writeGenerators(Network network, Map<Terminal, String> mapTe
                             cimNamespace, writer, context);
                     break;
                 case CgmesNames.EXTERNAL_NETWORK_INJECTION:
-                    String regulatingControlId = RegulatingControlEq.writeRegulatingControlEq(generator, exportedTerminalId(mapTerminal2Id, regulatingTerminal), regulatingControlsWritten, mode, cimNamespace, writer, context);
+                    if (context.isExportGeneratorsInLocalRegulationMode()) {
+                        regulatingControlId = RegulatingControlEq.writeRegulatingControlEq(generator, exportedTerminalId(mapTerminal2Id, generator.getTerminal()), regulatingControlsWritten, mode, cimNamespace, writer, context);
+                    } else {
+                        regulatingControlId = RegulatingControlEq.writeRegulatingControlEq(generator, exportedTerminalId(mapTerminal2Id, regulatingTerminal), regulatingControlsWritten, mode, cimNamespace, writer, context);
+                    }
```

Comment:
This code seems duplicated with the below case `case CgmesNames.SYNCHRONOUS_MACHINE:` maybe we can create a method to determine the `regulatingControlId` and use it in those 2 cases?

Reply:
I let the original code but instead set above the proper value for the regulatingTerminal depending on the regulation mode.

Reply:
Yes it seems good this way!